src/types.ts

```diff
@@ -7,7 +7,14 @@ export interface IdAndContentObject {
 
 export type RollupPluginSassOutputFn = (styles: string, styleNodes: IdAndContentObject[]) => any;
 
-export type RollupPluginSassProcessorFn<T = string | {css: string}> = (styles: string, id: string) => Promise<T> | T;
+export type RollupPluginSassProcessorFnOutput =
```

Comment:
I had to extract `RollupPluginSassProcessorFn` output to properly type usage on `transform` and `processRenderResponse` functions